### PR TITLE
[fix](value-range) fix the value range of non-nullable column contains null causes query short key index error.

### LINKED ISSUE #16502: [Bug] [daily] list partition table, select * where between and with wrong result
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master: doris-0.0.0-trunk-cf18de1

### What's Wrong?

jenkins failed case:
https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/201/testReport/(root)/test_sys_partition_list_alter/test_list_partition_select/

select * return 9 rows   and select k1 return 11 rows

image

### What You Expected?

return 11 rows

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #16502

## Problem summary

Querying short key indexes after code refactoring using MutableColumns to compare key values, when the key range of non-null columns contains null will be converted to the default value, which may cause errors.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.74 seconds
 stream load tsv:          467 seconds loaded 74807831229 Bytes, about 152 MB/s
 stream load json:         36 seconds loaded 2358488459 Bytes, about 62 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230223041549_clickbench_pr_102200.html

**Comment by luozenglin** [comment]
run p0

**Comment by luozenglin** [comment]
run compile

File: be/src/exec/olap_common.h [hunk 1]
```diff
@@ -312,13 +317,13 @@ class ColumnValueRange {
...
         _high_op = FILTER_LESS_OR_EQUAL;
-        _contain_null = true;
+        _contain_null = _is_nullable_col;
     }
 
     bool is_whole_value_range() const {
         return _fixed_values.empty() && _low_value == TYPE_MIN && _high_value == TYPE_MAX &&
                _low_op == FILTER_LARGER_OR_EQUAL && _high_op == FILTER_LESS_OR_EQUAL &&
-               contain_null();
+               _is_nullable_col == contain_null();
```
Comment (HappenLee) [inline be/src/exec/olap_common.h:326]: which case `_is_nullable_col` = false and  `contain_null()` = true

File: be/src/exec/olap_common.h [hunk 2]
```diff
@@ -386,8 +391,8 @@ class ColumnValueRange {
 
     static ColumnValueRange<primitive_type> create_empty_column_value_range(
             const std::string& col_name, int precision, int scale) {
-        return ColumnValueRange<primitive_type>(col_name, TYPE_MAX, TYPE_MIN, false, precision,
-                                                scale);
+        return ColumnValueRange<primitive_type>(col_name, TYPE_MAX, TYPE_MIN, true, false,
```
Comment (HappenLee) [inline be/src/exec/olap_common.h:394]: the function should be check is_nullable should be `true` ?

**Comment by luozenglin** [comment]
run buildall

**Comment by luozenglin** [comment]
run buildall

File: be/src/exec/olap_common.h (hunk 1 above)
Comment (luozenglin) [reply be/src/exec/olap_common.h:330]: I don't think this will happen, here add a DCHECK

File: be/src/exec/olap_common.h (hunk 2 above)
Comment (luozenglin) [reply be/src/exec/olap_common.h]: is_nullable_col has been passed in as a parameter

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**